rfcs/experimental/dynamic_selection/customization/custom_backends.md

```diff
@@ -0,0 +1,316 @@
+
+### Default Implementation Details
+
+The `backend_base` provides **minimal** default implementations for the core backend methods. Importantly:
+- `backend_base` does **not** support any reporting requirements by default
+- The default `scratch_t` template is empty (provides `no_scratch_t`)
+- Resource-specific features (timing, profiling, etc.) require specializing `core_resource_backend` for your `CoreResourceType`
+
+To create a backend with reporting support, specialize `core_resource_backend` for your specific `CoreResourceType` (e.g., `sycl::queue`).
+
+#### `submit` Implementation
+The default `submit` method calls the user-provided function with the unwrapped resource and returns a waitable submission type which wraps the return from the user-provided function.
+
+```cpp
+template <typename T>
+class default_submission
+{
+    T t;
+    void wait();
+    T unwrap();
+};
+
+template <typename SelectionHandle, typename Function, typename... Args>
+auto
+submit(SelectionHandle s, Function&& f, Args&&... args)
+{
+    return default_submission{std::forward<Function>(f)(oneapi::dpl::experimental::unwrap(s), std::forward<Args>(args)...)};
+}
+```
+**Assumptions**: The user function must be callable with a `ResourceType` resource and the `Function f` 
+returns an object that can be waited on. So if an adapter is provided, the user function is still called
+with the unadapted resource type. For example, if an adapter `[](auto pointer){ return *pointer; }` is used to
+reuse a `sycl::queue` with `sycl::queue *` resources, the user's function is still called with a 
+`sycl::queue *`.
+
+#### Instrumentation Support
+Where instrumentation is required, the customizer will need to override `submit` with code that performs instrumentation for the `CoreResourceType`. The overload must call the user function and return an object which supports `wait()` (blocks until the job completes) and `unwrap()` (returns the user's returned object).
+
+Each resource type has different characteristics for timing, profiling, and performance measurement, making a general instrumentation mechanism impossible.
+
+#### `get_resources()` Implementation
+Returns the vector of resources stored during construction. Resources must be copyable/movable into a `std::vector` and remain valid throughout the backend's lifetime.
+
+#### `get_submission_group()` Implementation
+Returns a group object that waits on all resources. The `default_submission_group` attempts to call `wait()` on the `CoreResourceType` by applying `adapter` to the `ResourceType` object. The `CoreResourceType` must provide a `wait()` method that blocks until all work on that resource is complete. Note that the default implementation waits on each resource, not each submission (works for SYCL queues, oneTBB `task_group` objects, etc.). Adapters can enable types without a `wait()` method; for example, `[](auto pointer){ return *pointer; }` allows `sycl::queue*` to work by adapting to `sycl::queue`.
```

Comment:
This is not completely accurate. get_submission_group should wait for all outstanding submissions made via this backend instance. One way to accomplish that is to wait on all resources, such as all queues. But we shouldn't require that it waits on the resources, only that all submissions are done.